scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/ProjectsController.java

```diff
@@ -0,0 +1,41 @@
+package edu.asu.diging.scriptoocloud.web;
+
+import java.util.Iterator;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.scriptoocloud.core.data.ProjectRepository;
+import edu.asu.diging.scriptoocloud.core.model.Project;
+import edu.asu.diging.scriptoocloud.core.model.impl.ProjectImpl;
+import edu.asu.diging.scriptoocloud.core.service.ProjectManager;
+
+@Controller
+public class ProjectsController {
+
+	@Autowired
+	private ProjectRepository projectRepo;
+
+	@Autowired
+	private RemoveProjectController removeProjectController;
+
+	@Autowired
+	private AddProjectController addProjectController;
```

Comment:
why is the above three classes autowired here?

Reply:
I brushed up on @Autowired documentation, and think my understanding was incorrect. 

I think the correct use is to remove the two @Autowired 's on the add/remove controllers, and keep them on the constructors within their respective classes. 

For the third one` @Autowired
    private ProjectRepository projectRepo;`, why would I remove this @Autowired if it is only defined here in the ProjectsController? My thinking is I need to autowire it since the add and remove controllers utilize the ProjectRepository. **Is Autowire only used for Controller dependencies?** Which would imply ProjectRepository is **not** a bean?

Reply:
You annotate a variable with `@Autowired` when you want to tell Spring that you need an instance of this interface. Here you need an instance of `ProjectManager` so you annotate it. This class does not need instances of the other controllers or the repository (where do you use them in the code?), so can delete line 24-31. It doesn't matter what a dependency is (controller, service, repository,...), just if you the class is managed by Spring and if you need an instance of it.  Does this make sense?

Reply:
Yes, thank you.